front-end/src/screens/AddressLoginForm/index.tsx

```diff
@@ -0,0 +1,218 @@
+// of the Apache-2.0 license. See the LICENSE file for details.
+
+import React, { useContext, useState } from 'react';
+import { Grid, DropdownProps } from 'semantic-ui-react';
+import styled from 'styled-components';
+import { InjectedAccountWithMeta } from '@polkadot/extension-inject/types';
+import { web3Accounts, web3FromSource, web3Enable } from '@polkadot/extension-dapp';
+import { stringToHex } from '@polkadot/util';
+
+import { NotificationContext } from '../../context/NotificationContext';
+import { useAddressLoginMutation, useAddressLoginStartMutation } from '../../generated/graphql';
+import ExtensionNotDetected from '../../components/ExtensionNotDetected';
+import { useRouter } from '../../hooks';
+import { handleLoginUser } from '../../services/auth.service';
+import Button from '../../ui-components/Button';
+import AccountSelectionForm from '../../ui-components/AccountSelectionForm';
+import FilteredError from '../../ui-components/FilteredError';
+import { Form } from '../../ui-components/Form';
+import { NotificationStatus } from '../../types';
+import getEncodedAddress from '../../util/getEncodedAddress';
+import cleanError from '../../util/cleanError';
+import { UserDetailsContext } from '../../context/UserDetailsContext';
+
+interface Props {
+	className?: string
+}
+
+const APPNAME = process.env.REACT_APP_APPNAME || 'polkassembly';
+
+const LoginForm = ({ className }:Props): JSX.Element => {
+	const [address, setAddress] = useState<string>('');
+	const [accounts, setAccounts] = useState<InjectedAccountWithMeta[]>([]);
+	const [extensionNotFound, setExtensionNotFound] = useState(false);
+	const [accountsNotFound, setAccountsNotFound] = useState(false);
+	const { history } = useRouter();
+	const { queueNotification } = useContext(NotificationContext);
+	const [addressLoginStartMutation] = useAddressLoginStartMutation();
+	const [addressLoginMutation, { loading, error }] = useAddressLoginMutation();
+	const currentUser = useContext(UserDetailsContext);
+
+	const getAccounts = async (): Promise<undefined> => {
+		const extensions = await web3Enable(APPNAME);
+
+		if (extensions.length === 0) {
+			setExtensionNotFound(true);
+			return;
+		} else {
+			setExtensionNotFound(false);
+		}
+
+		const accounts = await web3Accounts();
+
+		if (accounts.length === 0) {
+			setAccountsNotFound(true);
+			return;
+		} else {
+			setAccountsNotFound(false);
+		}
+
+		accounts.forEach((account) => {
+			account.address = getEncodedAddress(account.address) || account.address;
+		});
+
+		setAccounts(accounts);
+		if (accounts.length > 0) {
+			setAddress(accounts[0].address);
+		}
+
+		return;
+	};
+
+	const onAccountChange = (event: React.SyntheticEvent<HTMLElement, Event>, data: DropdownProps) => {
+		const addressValue = data.value as string;
+		setAddress(addressValue);
+	};
+
+	const navigateLogin = () => {
+		history.push('/login');
+	};
+
+	const handleLogin = async () => {
+		if (!accounts.length) {
+			return getAccounts();
```

Comment:
This should probably be in a useEffect() rather than in needing another click. I think we can ask for the extension permition right away, then let the user select the address etc..

Reply:
useEffect is buggy. Not loading accounts most of the time. Maybe due to web3 is not injected yet

Reply:
Can you show some code where it doesn't work well?

Reply:
change handleLogin to handleLogin and try to load the page. It will not show any accounts

Reply:
I don't get it, I placed 
```js
	useEffect(() => {
		if (!accounts.length) {
			getAccounts();
		}
	},[accounts.length]);
```

And that worked well. This way you get asked for the account when you land on the page, and then the account dopdown is shown. No need to click on "show accounts".

we should just make sure to have a spinner or something.

Reply:
i will add a delay to give page time to load web3

Reply:
Why? the useEffect just works for me.. I still don't get what you mean with giving "page time to load web3" we should show a loader with the text "waiting to get your accounts" or something, but that's it.

Reply:
ok